graphs/util/gpml_util_interface.py

```diff
@@ -0,0 +1,345 @@
+
+import graphspace_interface as interface
+import networkx as nx
+import json
+
+
+def parse_gpml(graph_json, title):
+    parse_json = {"graph": {"edges": [], "nodes": []}, "metadata": {}}
+    G = nx.DiGraph(directed=True)
+    # Use xml parser to parse the GPML file
+    import xml.etree.ElementTree as ET
+    tree = ET.fromstring(graph_json)
+    # The current GPML format
+    GPML = '{http://pathvisio.org/GPML/2013a}'
+    # Keep the overall count of nodes and create unique ids for
+    # datanodes, shape, labels.
+    # IMPORTANT!!!
+    count = 0
+    # Parsing the GPML file for Shape elements
+    shapes = tree.findall(GPML+'Shape')
+    G, parse_json, count = parse_shapes(G, parse_json, count, shapes)
+    # Find all the DataNodes in the GPML file which is used to draw various
+    # the various nodes of the graph
+    datanodes = tree.findall(GPML+'DataNode')
+    G, parse_json, count = parse_datanodes(G, parse_json, count, datanodes)
+    # Parsing the GPML file for Shape elements
+    labels = tree.findall(GPML+'Label')
+    G, parse_json, count = parse_labels(G, parse_json, count, labels)
+    # This is a crude version of getting interactions (edges) working
+    # Look for all the interactions
+    interactions = tree.findall(GPML+'Interaction')
+    G, parse_json, count = parse_interactions(G, parse_json, count, interactions)
+    print G.nodes(data=True)
+    layout = set_gpml_layout(G, parse_json)
+
+    # To add nodes to the group we parse through the group
```

Comment:
Change the comment to "To add nodes to the groups to which they belong, we parse the group ..."